src/main/java/com/cloudbees/jenkins/plugins/sshagent/SSHAgentStepExecution.java

```diff
@@ -17,24 +18,39 @@
 import javax.annotation.CheckReturnValue;
 import java.io.File;
 import java.io.IOException;
+import java.io.PrintStream;
 import java.io.PrintWriter;
 import java.io.StringWriter;
 import java.util.*;
+import java.util.logging.Level;
+import java.util.logging.Logger;
 
 public class SSHAgentStepExecution extends AbstractStepExecutionImpl {
 
     private static final long serialVersionUID = 1L;
 
-    @StepContextParameter
+    /**
+     * @deprecated use {@link #getListener()} instead
+     */
+    @Deprecated
     private transient TaskListener listener;
```

Comment:
I'm not 100% sure about this part, so please correct me. As I understand it we need to leave the fields so that ongoing builds are deserialized correctly, but removing StepContextParameter can be removed as it is nolonger needed?

Reply:
No. `transient` fields do not participate in deserialization. But the `@StepContextParameter` is inspected by the descriptor.